packages/special-pages/pages/onboarding/app/components/App.js

```diff
@@ -81,38 +71,55 @@ export function App ({ children }) {
     }
 
     /** @type {import('../types').Step['id'][]} */
-    const progress = ['privateByDefault', 'cleanerBrowsing', 'systemSettings', 'customize']
+    const progress = order.slice(2, -1)
     const showProgress = progress.includes(activeStep)
 
+    function animationDidFinish (e) {
+        if (e.target?.dataset?.exiting === 'true') {
+            advance()
+        }
+    }
+
     return (
         <main className={styles.main}>
-            <Background />
-            {debugState && <Debug state={globalState} />}
+            <link rel="preload" href={['js', animation].join('/')} as="image"/>
+            <link rel="preload" href={['js', stepMeta.dockSingle.rows.dock.path].join('/')} as="image"/>
+            <link rel="preload" href={['js', stepMeta.importSingle.rows.import.path].join('/')} as="image"/>
+            <link rel="preload" href={['js', stepMeta.makeDefaultSingle.rows['default-browser'].path].join('/')} as="image"/>
+            <Background/>
+            {debugState && <Debug state={globalState}/>}
             <div className={styles.container} data-current={activeStep}>
-                <ErrorBoundary didCatch={didCatch} fallback={<Fallback />}>
-                    {/* This is used to allow an 'exit' animation to take place */}
-                    {exiting && <Timeout onComplete={next} timeout={['welcome', 'getStarted'].includes(activeStep) ? 0 : 600} />}
+                <ErrorBoundary didCatch={didCatch} fallback={<Fallback/>}>
                     <Stack>
-                        <Header aside={showProgress && <Progress current={progress.indexOf(activeStep) + 1} total={progress.length} />}>
+                        <Header aside={showProgress && <Progress current={progress.indexOf(activeStep) + 1} total={progress.length}/>}>
                             <Typed
                                 onComplete={titleDone}
                                 text={pageTitle}
                                 data-current={activeStep}
-                                data-exiting={String(exiting)}
+                                data-exiting={pageTitle !== nextPageTitle && String(exiting)}
```

Comment:
we only allow the 'exiting' state (animating up) if the next title differs - this allows us to prevent the exact same title from animating in when it's the same as the previous step